File: RPICommunicator_V1/app/src/main/java/com/example/rpicommunicator_v1/Communication/SendThread.java
package com.example.rpicommunicator_v1.Communication;

import android.util.Log;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.net.InetAddress;
import java.net.Socket;
import java.net.UnknownHostException;

import static com.example.rpicommunicator_v1.ViewAndModels.Constants.IP;
import static com.example.rpicommunicator_v1.ViewAndModels.Constants.PORT;

public class SendThread extends Thread{

    private final String message;
    private  InetAddress localIP;

    public SendThread(String message, String localIP) {
        this.message = message;
        try {
            this.localIP = InetAddress.getByName(localIP);
        } catch (UnknownHostException e) {
            try {
                this.localIP = InetAddress.getByName(IP);
            } catch (UnknownHostException unknownHostException) {
                unknownHostException.printStackTrace();
            }
            e.printStackTrace();
        }
    }

    public SendThread(String message) {
        this.message = message;
    }

    public void run() {
        Log.i("Communicator","Message: "+message);
        Socket socket = null;
        try {
            if(localIP==null) {
                socket = new Socket(IP, PORT);
            }else{
                socket=new Socket(localIP, PORT);
            }
            DataOutputStream dout = new DataOutputStream(socket.getOutputStream());
            DataInputStream din = new DataInputStream(socket.getInputStream());

            dout.writeUTF(message);
            dout.flush();


            Log.i("run", message + " sent");
            String str = din.readUTF();//in.readLine();

            Log.i("run", "Reply: " + str);


            dout.close();
            din.close();
            socket.close();
            //
            // Toast.makeText(context, text+" was successful", Toast.LENGTH_LONG).show();
            Log.i("run", "Thread terminated");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/example/rpicommunicator_v1/service/SendThread.kt
package com.example.rpicommunicator_v1.service

import android.app.Activity
import android.os.StrictMode
import android.os.StrictMode.ThreadPolicy
import android.util.Log
import com.example.rpicommunicator_v1.component.Constants.IP
import com.example.rpicommunicator_v1.component.Constants.PORT
import java.io.DataInputStream
import java.io.DataOutputStream
import java.io.IOException
import java.net.InetAddress
import java.net.Socket
import java.net.UnknownHostException


class SendThread
    (private val message: String, localIP: String, localPort: Int) : Thread() {
    private var localIP: InetAddress? = null
    private var localPort: Int = PORT
    private var activity: Activity? = null

    init {
        val policy = ThreadPolicy.Builder().permitAll().build()
        StrictMode.setThreadPolicy(policy)
        this.localPort = localPort
        try {
            this.localIP = InetAddress.getByName(localIP)
        } catch (e: UnknownHostException) {
            try {
                this.localIP = InetAddress.getByName(IP)
            } catch (unknownHostException: UnknownHostException) {
                unknownHostException.printStackTrace()
            }
            e.printStackTrace()
        }
    }


    override fun run() {
        val socket: Socket
        try {
            if (localIP == null) {
                socket = Socket(IP, localPort)
            } else {
                socket = Socket(localIP, localPort)
            }
            val dout = DataOutputStream(socket.getOutputStream())
            val din = DataInputStream(socket.getInputStream())
            dout.writeUTF(message)
            dout.flush()
            val str = din.readUTF()

            Log.i("run", "Reply: $str")
            dout.close()
            din.close()
            socket.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}